fix(load-balancer): add new info and reorganise existing (#934)
scaleway/docs-content: network/load-balancer/quickstart.mdx

  h1: Load Balancers - Quickstart
  paragraph: This page shows you how to get started with Scaleway Load Balancers.
steps:
dates:
  validation: 2022-11-03
  posted: 2021-05-26
---

[Load Balancers](https://www.scaleway.com/en/load-balancer/) are highly available and fully-managed Instances which allow you to distribute your workload across your various servers. They ensure the scaling of all your applications while securing their continuous availability, even in the event of heavy traffic. Load Balancers are commonly used to improve the performance and reliability of web sites, applications, databases and other services.

> <Lightbox src="scaleway-load-balancer.webp" alt="" />

Reviewer: I don't think this screenshot is appropriate since it's a screenshot from a Kapsule LB with a big red warning in it. Not really sexy :)

Author: No it's still there, just as above!
If you want, you could do a `yarn clean && yarn && yarn build && yarn serve`, which takes longer than a yarn develop, but builds the whole site more thoroughly. Once it's run, go to `http://localhost:9000/en/docs/` to view the build of the entire site, and that might solve the problel you're having with images.

<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com) 
</Message>

## How to create a Load Balancer

1. Click **Load Balancers** in the **Network** section of the [Scaleway Console](https://console.scaleway.com) side menu. If you have not already created a Load Balancer, the product creation page is displayed.

2. Click **Create a Load Balancer**. The creation wizard displays.

> 3. Enter a **name** and **description** for your Load Balancer. Optionally, you can assign **tags** to help organize your Load Balancers.

Reviewer: The description field is optional, do you think we should mention it in the second sentence instead?

Author: yes, done

    <Lightbox src="scaleway-load-balancer-name.webp" alt="" />

> 4. Choose the **Availability Zone** in which your Load Balancer will be deployed geographically. Currently we provide the following Availability Zones:

Reviewer: Can you please update the AZ list as we have already added AMS2 since this document was last updated. I also think we should rename Warsaw to Warsaw 1 since this is how it is displayed in the console now.
Please note we will have to update this page really soon with WAW2 availability.

Author: done

      - `PAR1`: Paris 1, France.
      - `PAR2`: Paris 2, France (innovative and sustainable availability zone).
      - `AMS1`: Amsterdam, The Netherlands.
      - `WAW`: Warsaw, Poland.

5. Select a Load Balancer **type**, according to your bandwidth needs.
    <Lightbox src="scaleway-select-load-balancer.webp" alt="" />

6. Select an **IP address** for your Load Balancer. You can either allocate a new IP, or select one of your existing IPs.

7. Configure your Load Balancer's **[frontend](/network/load-balancer/concepts/#frontends)** by entering a **name** and a **port**. The frontend will listen on this port, and forward the requests it receives from clients to the backend(s). Note that you cannot add an SSL certificate at this stage, but you can [add one after the creation of the Load Balancer](/network/load-balancer/how-to/add-certificate/). You can also add [additional frontends](#how-to-add-frontends) after creation.
    <Lightbox src="scaleway-configure-frontend.webp" alt="" />

> 8. Configure your Load Balancer's **[backend](/network/load-balancer/concepts/#backends)** by entering a **name**, and the **protocol** and **port** it should use to connect to the backend servers. Enter also a **proxy**, **health check** method and the **server IPs** of the backend servers it should direct traffic to. You can [add additional backends](#how-to-add-backends) after creating your Load Balancer, if required.

Reviewer: The mention to proxy is not really clear. It would be better to say something like "select a proxy protocol option"

> 8. Configure your Load Balancer's **[backend](/network/load-balancer/concepts/#backends)** by entering a **name**, and the **protocol** and **port** it should use to connect to the backend servers. Enter also a **proxy**, **health check** method and the **server IPs** of the backend servers it should direct traffic to. You can [add additional backends](#how-to-add-backends) after creating your Load Balancer, if required.

Reviewer: > "and the server IPs of the backend servers it should direct traffic to"

I would say "add the IP addresses (one or many) of the servers the backend must forward traffic to".

Author: done (slightly different wording)

    <Lightbox src="scaleway-configure-backend.webp" alt="" /> 
    Depending on the options you choose, you may be prompted for additional configurations such as the **Verify certificate** setting if you select the HTTPS protocol. For more information on choosing configuration settings for your Load Balancer's backend, see our [dedicated reference documentation](/network/load-balancer/reference-content/configuring-load-balancer)

9. Configure **advanced settings** if required:
    - **S3 failover**: activate the feature and enter a Scaleway Object Storage [Bucket Website](/storage/object/concepts/#bucket-website) URL. If none of your backend servers are available, users will be redirected to this page. 
      <Message type="note">
        The S3 failover feature is only available for Load Balancers using the HTTP protocol on their backend. Find out more about setting up a static website with Object Storage in our [dedicated documentation](/storage/object/how-to/use-bucket-website/).
      </Message>

>     - **[Sticky session](/network/load-balancer/concepts/#sticky-session)**: activate the feature and (if relevant) enter the name for a cookie that will bind users' sessions to specific backend servers.

Reviewer: Maybe a bit more info could help here like: "Use this feature if you want a particular user session to stick to the same backend server. This is useful if your users need their requests to be served by a single server in order to maintain a consistent session".

Author: done

1. Click **Load Balancers** in the **Network** section of the [Scaleway Console](https://console.scaleway.com) side menu. The Load Balancer dashboard displays. 
2. Click the Load Balancer you want to add frontends to. The Load Balancer information page displays. 
3. Click the **Frontends** tab.
4. Click the green **+** button to launch the frontend creation wizard.
    <Lightbox src="scaleway-create-frontend.webp" alt="" />
5. Enter a **name** and **port** for your new frontend. Note that each frontend must listen on a different port.
6. Select an [**SSL certificate(s)**](/network/load-balancer/how-to/add-certificate) for your frontend to use, if required. Note that you must already have [created or imported an SSL certificate for your Load Balancer](/network/load-balancer/how-to/add-certificate/) in order to select one here.
6. Choose between **using an existing backend** or **creating a new backend**. Your frontend will direct traffic to the backend you choose here.
    - If using an existing backend, click **Use an existing backend** and select the backend of your choice from the dropdown list.

>     - If creating a new backend, click **Create a new backend** and enter a name, protocol, port proxy, health check type, server IP(s) and advanced settings as required. See our [dedicated documentation](/network/load-balancer/reference-content/configuring-load-balancer/) for help configuring your backend

Reviewer: > enter a name, protocol, port proxy, health check type, server IP(s) and advanced settings as required

enter a name, protocol, port, proxy protocol, health check type, server IP(s) and advanced settings as required

Proxy alone is confusing, we are selecting a proxy protocol option here.

7. Click **Create Frontend**.

## How to add backends

> You can add more backends to your Load Balancer after its creation. Each backend can be connected to a different frontend and different backend servers as required, allowing you to route traffic through your Load Balancer according to your needs.

Reviewer: > Each backend can be connected to a different frontend and different backend servers as required, allowing you to route traffic through your Load Balancer according to your needs

This sentence is a bit confusing, I am not sure to understand what we are trying to say.

Author: I've modified it, OK now?

Reviewer: Much better, thanks.

1. Click **Load Balancers** in the **Network** section of the [Scaleway Console](https://console.scaleway.com) side menu. The Load Balancer dashboard displays. 
2. Click the Load Balancer you want to add backends to. The Load Balancer information page displays. 
3. Click the **Backends** tab.
4. Click the green **+** button.
    <Lightbox src="scaleway-create-backend.webp" alt="" />

> 5. Enter a name, protocol, port, proxy, health check type,  server IP(s) and advanced settings as required for your new backend. See our [dedicated page](/network/load-balancer/reference-content/configuring-load-balancer/) for help configuring your backend, including details of how to set up SSL bridging, offloading or passthrough.

Reviewer: > Enter a name, protocol, port, proxy, health check type

enter a name, protocol, port, proxy protocol, health check type, server IP(s) and advanced settings as required

Proxy alone is confusing, we are selecting a proxy protocol option here.

Author: done

    - If using an existing backend, click **Use an existing backend** and select the backend of your choice from the dropdown list.
    - If creating a new backend, click **Create a new backend** and enter a name, port, protocol, proxy protocol, health check type, server IP(s) and advanced settings as required. See our [dedicated documentation](/network/load-balancer/reference-content/configuring-load-balancer/) for help configuring your backend
7. Click **Create Frontend**.

## How to add backends

You can add more backends to your Load Balancer after its creation. Each backend can be connected to a different frontend, allowing you to route traffic through your Load Balancer according to your needs.

1. Click **Load Balancers** in the **Network** section of the [Scaleway Console](https://console.scaleway.com) side menu. The Load Balancer dashboard displays. 
2. Click the Load Balancer you want to add backends to. The Load Balancer information page displays. 
3. Click the **Backends** tab.
4. Click the green **+** button.
    <Lightbox src="scaleway-create-backend.webp" alt="" />

> 5. Enter a name, protocol, port,  protocol, health check type,  server IP(s) and advanced settings as required for your new backend. See our [dedicated page](/network/load-balancer/reference-content/configuring-load-balancer/) for help configuring your backend, including details of how to set up SSL bridging, offloading or passthrough.

Reviewer 2: ```suggestion
5. Enter a name, protocol, port, protocol, health check type,  server IP(s) and advanced settings as required for your new backend. See our [dedicated page](/network/load-balancer/reference-content/configuring-load-balancer/) for help configuring your backend, including details of how to set up SSL bridging, offloading or passthrough.
```

7. Configure your Load Balancer's **[frontend](/network/load-balancer/concepts/#frontends)** by entering a **name** and a **port**. The frontend will listen on this port, and forward the requests it receives from clients to the backend(s). Note that you cannot add an SSL certificate at this stage, but you can [add one after the creation of the Load Balancer](/network/load-balancer/how-to/add-certificate/). You can also add [additional frontends](#how-to-add-frontends) after creation.
    <Lightbox src="scaleway-configure-frontend.webp" alt="" />

8. Configure your Load Balancer's **[backend](/network/load-balancer/concepts/#backends)** by entering a **name**, and the **protocol** and **port** it should use to connect to the backend servers. You should also a select a **proxy** protocol option, **health check** method and add the **server IPs** (one or many) of the backend server(s) it should direct traffic to. You can [add additional backends](#how-to-add-backends) after creating your Load Balancer, if required.
    <Lightbox src="scaleway-configure-backend.webp" alt="" /> 

>     Depending on the options you choose, you may be prompted for additional configurations such as the **Verify certificate** setting if you select the HTTPS protocol. For more information on choosing configuration settings for your Load Balancer's backend, see our [dedicated reference documentation](/network/load-balancer/reference-content/configuring-load-balancer)

Reviewer 3: ```suggestion
    Depending on the options you choose, you may be prompted for additional configurations such as the **Verify certificate** setting if you select the HTTPS protocol. For more information on choosing configuration settings for your Load Balancer's backend, see our [dedicated reference documentation](/network/load-balancer/reference-content/configuring-load-balancer).
```

10. Click **Create a Load Balancer**.

    <Message type="note">
    * By default, each Load Balancer is created using [round-robin](/network/load-balancer/concepts/#balancing-rules) balancing rules.
    * You can [add additional frontends and backends](/network/load-balancer/how-to/manage-frontends-and-backends/) to your Load Balancer after its creation.
    </Message>

## How to add frontends

You can add more frontends to your Load Balancer after its creation. This allows you to configure different frontends listening on different ports, and to connect different frontends to different backends as required.

1. Click **Load Balancers** in the **Network** section of the [Scaleway Console](https://console.scaleway.com) side menu. The Load Balancer dashboard displays. 
2. Click the Load Balancer you want to add frontends to. The Load Balancer information page displays. 
3. Click the **Frontends** tab.

> 4. Click the green **+** button to launch the frontend creation wizard.

Reviewer 3: ```suggestion
4. Click <Icon name="plus" /> to launch the frontend creation wizard.
```

    <Lightbox src="scaleway-create-frontend.webp" alt="" />
5. Enter a **name** and **port** for your new frontend. Note that each frontend must listen on a different port.
6. Select an [**SSL certificate(s)**](/network/load-balancer/how-to/add-certificate) for your frontend to use, if required. Note that you must already have [created or imported an SSL certificate for your Load Balancer](/network/load-balancer/how-to/add-certificate/) in order to select one here.
6. Choose between **using an existing backend** or **creating a new backend**. Your frontend will direct traffic to the backend you choose here.
    - If using an existing backend, click **Use an existing backend** and select the backend of your choice from the dropdown list.

>     - If creating a new backend, click **Create a new backend** and enter a name, port, protocol, proxy protocol, health check type, server IP(s) and advanced settings as required. See our [dedicated documentation](/network/load-balancer/reference-content/configuring-load-balancer/) for help configuring your backend

Reviewer 3: ```suggestion
    - If creating a new backend, click **Create a new backend** and enter a name, port, protocol, proxy protocol, health check type, server IP(s) and advanced settings as required. See our [dedicated documentation](/network/load-balancer/reference-content/configuring-load-balancer/) for help configuring your backend.
```
